### PR TITLE
VZ-7380: Update application-operator clusters API content

### PR DESCRIPTION
**Pull request by bobdonat**
This pull request updates the application-operator clusters API content for auto-generation of the API doc.

### CODE REVIEW

File: application-operator/apis/clusters/v1alpha1/multiclusterapplicationconfiguration_types.go [hunk 1]
```diff
@@ -11,43 +11,47 @@ import (
...
+	// +optional
 	Secrets []string `json:"secrets,omitempty"`
+
+	// Template containing the metadata and spec for an OAM applicationConfiguration resource.
+	Template ApplicationConfigurationTemplate `json:"template"`
 }
 
-// ApplicationConfigurationTemplate has the metadata and spec of the underlying
-// OAM ApplicationConfiguration
+// ApplicationConfigurationTemplate has the metadata and embedded spec of the OAM applicationConfiguration resource.
```
Comment (MelwynPaul) [inline application-operator/apis/clusters/v1alpha1/multiclusterapplicationconfiguration_types.go:28]: spec > specification

File: application-operator/apis/clusters/v1alpha1/multiclustercomponent_types.go [hunk 2]
```diff
@@ -11,39 +11,43 @@ import (
...
-	// Clusters in which the secret is to be placed
+	// Clusters in which the component is to be created.
 	Placement Placement `json:"placement"`
+
+	// Template containing the metadata and spec for an OAM component.
+	Template ComponentTemplate `json:"template"`
 }
 
-// ComponentTemplate has the metadata and spec of the underlying OAM component
+// ComponentTemplate has the metadata and embedded spec of the OAM component.
```
Comment (MelwynPaul) [inline application-operator/apis/clusters/v1alpha1/multiclustercomponent_types.go:23]: spec > specification

File: application-operator/apis/clusters/v1alpha1/multiclusterconfigmap_types.go [hunk 3]
```diff
@@ -10,48 +10,52 @@ import (
...
+	// Clusters in which the config map is to be created.
 	Placement Placement `json:"placement"`
+
+	// The embedded Kubernetes config map.
+	Template ConfigMapTemplate `json:"template"`
 }
 
-// ConfigMapTemplate has the metadata and spec of the underlying ConfigMap
-// Note that K8S does not define a "ConfigMapSpec" data type, so fields in ConfigMap are copied here
+// ConfigMapTemplate has the metadata and spec of the Kubernetes ConfigMap.
```
Comment (MelwynPaul) [inline application-operator/apis/clusters/v1alpha1/multiclusterconfigmap_types.go:22]: spec > specification

File: application-operator/apis/clusters/v1alpha1/multiclustersecret_types.go [hunk 4]
```diff
@@ -11,48 +11,51 @@ import (
...
+	// Clusters in which the secret is to be created.
 	Placement Placement `json:"placement"`
+
+	// The embedded Kubernetes secret.
+	Template SecretTemplate `json:"template"`
 }
 
-// SecretTemplate has the metadata and spec of the underlying secret
-// Note that K8S does not define a "SecretSpec" data type, so the 3 fields in Secret are copied here
+// SecretTemplate has the metadata and spec of the Kubernetes Secret.
```
Comment (MelwynPaul) [inline application-operator/apis/clusters/v1alpha1/multiclustersecret_types.go:23]: spec > specification

File: platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclusterapplicationconfigurations.yaml [hunk 5]
```diff
@@ -416,7 +415,11 @@ spec:
                   type: object
                 type: array
               state:
-                description: State of the multi cluster resource
+                description: 'The state of the multicluster resource. State values
+                  are case-sensitive and formatted as follows: <ul><li>`Failed`: rdeployment
```
Comment (MelwynPaul) [inline platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclusterapplicationconfigurations.yaml:419]: Typo: rdeployment > deployment

File: platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclustercomponents.yaml [hunk 6]
```diff
@@ -197,7 +194,11 @@ spec:
                   type: object
                 type: array
               state:
-                description: State of the multi cluster resource
+                description: 'The state of the multicluster resource. State values
+                  are case-sensitive and formatted as follows: <ul><li>`Failed`: rdeployment
```
Comment (MelwynPaul) [inline platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclustercomponents.yaml:198]: Typo: rdeployment > deployment

File: platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclusterconfigmaps.yaml [hunk 7]
```diff
@@ -160,7 +157,11 @@ spec:
                   type: object
                 type: array
               state:
-                description: State of the multi cluster resource
+                description: 'The state of the multicluster resource. State values
+                  are case-sensitive and formatted as follows: <ul><li>`Failed`: rdeployment
```
Comment (MelwynPaul) [inline platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclusterconfigmaps.yaml:161]: Typo: rdeployment > deployment

File: platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclustersecrets.yaml [hunk 8]
```diff
@@ -159,7 +156,11 @@ spec:
                   type: object
                 type: array
               state:
-                description: State of the multi cluster resource
+                description: 'The state of the multicluster resource. State values
+                  are case-sensitive and formatted as follows: <ul><li>`Failed`: rdeployment
```
Comment (MelwynPaul) [inline platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclustersecrets.yaml:160]: rdeployment > deployment

File: platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_verrazzanoprojects.yaml [hunk 9]
```diff
@@ -775,7 +774,11 @@ spec:
                   type: object
                 type: array
               state:
-                description: State of the multi cluster resource
+                description: 'The state of the multicluster resource. State values
+                  are case-sensitive and formatted as follows: <ul><li>`Failed`: rdeployment
```
Comment (MelwynPaul) [inline platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_verrazzanoprojects.yaml:778]: rdeployment > deployment

**Review by MelwynPaul — COMMENTED** [review]
Looks good. 
Added a few comments.

File: application-operator/apis/clusters/v1alpha1/multicluster_common_types.go [hunk 10]
```diff
@@ -7,99 +7,100 @@ import corev1 "k8s.io/api/core/v1"
...
-// ConditionType identifies the condition of the multi-cluster resource which can be checked with kubectl wait
+// ConditionType identifies the condition of the multi-cluster resource which can be checked with `kubectl wait`.
 type ConditionType string
 
 const (
-	// DeployPending means deployment to specified cluster is in progress.
-	DeployPending ConditionType = "DeployPending"
-
-	// DeployComplete means deployment to specified cluster completed successfully
+	// DeployComplete means deployment to specified cluster completed successfully.
```
[the remark this answers is not in the document]
Comment (bobdonat) [reply application-operator/apis/clusters/v1alpha1/multicluster_common_types.go]: fixed

File: application-operator/apis/clusters/v1alpha1/multicluster_common_types.go [hunk 11]
```diff
@@ -7,99 +7,100 @@ import corev1 "k8s.io/api/core/v1"
...
+	DeployPending ConditionType = "DeployPending"
 )
 
-// StateType identifies the state of a multi-cluster resource
+// StateType identifies the state of a multi-cluster resource.
 type StateType string
 
 const (
-	// Pending is the state when deploy to specified cluster is in progress
+	// Failed is the state when deploy to specified cluster has failed.
```
[the remark this answers is not in the document]
Comment (bobdonat) [reply application-operator/apis/clusters/v1alpha1/multicluster_common_types.go:66]: fixed

File: application-operator/apis/clusters/v1alpha1/multicluster_common_types.go [hunk 12]
```diff
@@ -7,99 +7,100 @@ import corev1 "k8s.io/api/core/v1"
...
+	Failed StateType = "Failed"
+
+	// Pending is the state when deploy to specified cluster is in progress.
```
[the remark this answers is not in the document]
Comment (bobdonat) [reply application-operator/apis/clusters/v1alpha1/multicluster_common_types.go:69]: fixed

File: application-operator/apis/clusters/v1alpha1/multicluster_common_types.go [hunk 13]
```diff
@@ -7,99 +7,100 @@ import corev1 "k8s.io/api/core/v1"
...
 	Pending StateType = "Pending"
 
-	// Succeeded is the state when deploy to specified cluster is completed
+	// Succeeded is the state when deploy to specified cluster is completed.
```
[the remark this answers is not in the document]
Comment (bobdonat) [reply application-operator/apis/clusters/v1alpha1/multicluster_common_types.go:72]: fixed

File: application-operator/apis/clusters/v1alpha1/multicluster_common_types.go [hunk 14]
```diff
@@ -7,99 +7,100 @@ import corev1 "k8s.io/api/core/v1"
...
-	// The latest available observations of an object's current state.
+	// Status information for each cluster.
+	Clusters []ClusterLevelStatus `json:"clusters,omitempty"`
+
+	// The current state of a multicluster resource.
 	Conditions []Condition `json:"conditions,omitempty"`
 
-	// State of the multi cluster resource
+	// The state of the multicluster resource. State values are case-sensitive and formatted as follows:
+	// <ul><li>`Failed`: rdeployment to cluster failed</li><li>`Pending`: deployment to cluster is in progress</li><li>`Succeeded`: deployment to cluster successfully completed</li></ul>
```
[the remark this answers is not in the document]
Comment (bobdonat) [reply application-operator/apis/clusters/v1alpha1/multicluster_common_types.go]: fixed

File: application-operator/apis/clusters/v1alpha1/multiclusterapplicationconfiguration_types.go (hunk 1 above)
Comment (bobdonat) [reply application-operator/apis/clusters/v1alpha1/multiclusterapplicationconfiguration_types.go:28]: spec is the name of the field so left as is

File: application-operator/apis/clusters/v1alpha1/multiclustercomponent_types.go (hunk 2 above)
Comment (bobdonat) [reply application-operator/apis/clusters/v1alpha1/multiclustercomponent_types.go:23]: spec is the name of the field so left as is

File: application-operator/apis/clusters/v1alpha1/multiclusterconfigmap_types.go (hunk 3 above)
Comment (bobdonat) [reply application-operator/apis/clusters/v1alpha1/multiclusterconfigmap_types.go:22]: spec is the name of the field so left as is

File: application-operator/apis/clusters/v1alpha1/multiclustersecret_types.go (hunk 4 above)
Comment (bobdonat) [reply application-operator/apis/clusters/v1alpha1/multiclustersecret_types.go:23]: spec is the name of the field so left as is

File: platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclusterapplicationconfigurations.yaml (hunk 5 above)
Comment (bobdonat) [reply platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclusterapplicationconfigurations.yaml]: fixed

File: platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclustercomponents.yaml (hunk 6 above)
Comment (bobdonat) [reply platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclustercomponents.yaml]: fixed

File: platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclusterconfigmaps.yaml (hunk 7 above)
Comment (bobdonat) [reply platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_multiclusterconfigmaps.yaml]: fixed

File: platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_verrazzanoprojects.yaml (hunk 9 above)
Comment (bobdonat) [reply platform-operator/helm_config/charts/verrazzano-application-operator/crds/clusters.verrazzano.io_verrazzanoprojects.yaml]: fixed

**Review by MelwynPaul — APPROVED** [review]
Looks good. 
Approved.

**Pull request merged**